src/libraries/Microsoft.Extensions.Options/tests/Microsoft.Extensions.Options.Tests/OptionsBuilderTest.cs

```diff
@@ -665,5 +665,125 @@ public void CanValidateMixDataAnnotations()
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        public void ValidateOnStart_CallValidateOnStartFirst_ValidationSuccessful()
+        {
+            var services = new ServiceCollection();
+
+            services.AddOptions<AnnotatedOptions>()
+                    .ValidateOnStart()
+                    .Configure(o =>
+                    {
+                        o.StringLength = "111111";
+                        o.IntRange = 10;
+                        o.Custom = "nowhere";
+                        o.Dep1 = "Not dep2";
+                    })
+                    .ValidateDataAnnotations()
+                    .Validate(o => o.Custom != "nowhere", "I don't want to go to nowhere!");
+
+            var sp = services.BuildServiceProvider();
+
+            var error = Assert.Throws<OptionsValidationException>(() => sp.GetRequiredService<IOptions<AnnotatedOptions>>().Value);
+            ValidateFailure<AnnotatedOptions>(error, Options.DefaultName, 6,
+                    "DataAnnotation validation failed for members: 'Required' with the error: 'The Required field is required.'.",
+                    "DataAnnotation validation failed for members: 'StringLength' with the error: 'Too long.'.",
+                    "DataAnnotation validation failed for members: 'IntRange' with the error: 'Out of range.'.",
+                    "DataAnnotation validation failed for members: 'Custom' with the error: 'The field Custom is invalid.'.",
+                    "DataAnnotation validation failed for members: 'Dep1,Dep2' with the error: 'Dep1 != Dep2'.",
+                    "I don't want to go to nowhere!");
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        public void ValidateOnStart_ConfigureBasedOnDataAnnotationRestrictions_ValidationSuccessful()
+        {
+            var services = new ServiceCollection();
+            services.AddOptions<AnnotatedOptions>()
+                .Configure(o =>
+                {
+                    o.Required = "required";
+                    o.StringLength = "1111";
+                    o.IntRange = 0;
+                    o.Custom = "USA";
+                    o.Dep1 = "dep";
+                    o.Dep2 = "dep";
+                })
+                .ValidateDataAnnotations()
+                .ValidateOnStart()
+                .Validate(o => o.Custom != "nowhere", "I don't want to go to nowhere!");
+
+            var sp = services.BuildServiceProvider();
+
+            try
+            {
+                var value = sp.GetRequiredService<IOptions<AnnotatedOptions>>().Value;
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
```

Comment:
Why are we catching and asserting here? Why not just take the `try` out and let the test throw if it does?